contracts/resolvers/DelegatableResolver.sol

```diff
@@ -0,0 +1,126 @@
+pragma solidity >=0.8.4;
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import "./profiles/ABIResolver.sol";
+import "./profiles/AddrResolver.sol";
+import "./profiles/ContentHashResolver.sol";
+import "./profiles/DNSResolver.sol";
+import "./profiles/InterfaceResolver.sol";
+import "./profiles/NameResolver.sol";
+import "./profiles/PubkeyResolver.sol";
+import "./profiles/TextResolver.sol";
+import "./profiles/ExtendedResolver.sol";
+import "./Multicallable.sol";
+import "./IDelegatableResolver.sol";
+
+/**
+ * A delegated resolver that allows the resolver owner to add an operator to update records of a node on behalf of the owner.
+ * address.
+ */
+contract DelegatableResolver is
+    Ownable,
+    Multicallable,
+    ABIResolver,
+    AddrResolver,
+    ContentHashResolver,
+    DNSResolver,
+    InterfaceResolver,
+    NameResolver,
+    PubkeyResolver,
+    TextResolver,
+    ExtendedResolver
+{
+    using BytesUtils for bytes;
+
+    // Logged when an operator is added or removed.
+    event Approval(
+        bytes32 indexed node,
+        address indexed operator,
+        bytes name,
+        bool approved
+    );
+
+    error NotAuthorized(bytes32 node);
+
+    constructor(address owner) {
+        operators[bytes32(0)][owner] = true;
```

Comment:
What's the purpose of this?

Reply:
It makes the owner the owner of the root node on this resolver, which allows them to update records anywhere in the hierarchy.

Reply:
This part will be replaced by the factory PR https://github.com/ensdomains/ens-contracts/pull/289